### ISSUE TITLE
docs(skore): Document behaviour of `CrossValidationReport.cache_predictions`

### ISSUE THREAD

**Issue by augustebaum** [probabl-ai/skore#1162]
The following discussion should be distilled in the docs somewhere, either in the `cache_predictions` docstring or in the `CrossValidationReport` docstring.

---

`cache_predictions` doesn't seem to cache things:
```
>>> from sklearn.datasets import make_classification
>>> from sklearn.model_selection import train_test_split
>>> from sklearn.linear_model import LogisticRegression
>>> X, y = make_classification(random_state=42)
>>> estimator = LogisticRegression()
>>> from skore import CrossValidationReport
>>> report = CrossValidationReport(estimator, X=X, y=y, cv_splitter=2)

  Processing cross-validation ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%
  for LogisticRegression

>>> report._cache
{}

>>> report.cache_predictions()
  Cross-validation predictions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%
  Caching predictions          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%
  Caching predictions          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100%

>>> report._cache
{}
```
What am I getting wrong?

_Originally posted by @augustebaum in https://github.com/probabl-ai/skore/pull/1091#discussion_r1920230535_

---

Answer by @glemaitre:
> Nop you are right but it is normal here, because the cache_predictions are cached in each individual EstimatorReport under the hood. So we don't need to cache anything in the cache of the CrossValidationReport.
> 
> Trigger a metric and in this case, the aggregate metric is cached at the level of the CrossValidationReport.

> To illustrate:

```python
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
X, y = make_classification(random_state=42)
estimator = LogisticRegression()
from skore import CrossValidationReport
report = CrossValidationReport(estimator, X=X, y=y, cv_splitter=2)
report.cache_predictions()
print(report.estimator_reports_[0]._cache)
```

```
{(np.int64(710796380309702198),
  'predict',
  'test'): array([0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0,
        0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1,
        1, 1, 1, 1, 1, 1]),
 (np.int64(710796380309702198),
  'predict',
  'train'): array([0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0,
        1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1,
        1, 0, 1, 1, 0, 0]),
 (np.int64(710796380309702198),
  np.int64(0),
  'predict_proba',
...
}
```

```python
# new call
report.metrics.accuracy()
report._cache
```

```
{(np.int64(-3970494336380670746),
  'accuracy',
  'test',
  None): Metric                       Accuracy (↗︎)
 LogisticRegression Split #0            1.0
                    Split #1            0.9}

```

**Comment by glemaitre** [comment]
Thinking a bit more, I don't think we should document by showing the `_cache`. A user should not go check those. The example written here: https://github.com/probabl-ai/skore/pull/1156/files#diff-c479e9f225f43b67c50708409202205c8391bb393335e1cae99093a20bf95921 provides some details.

So I would be more in favor of calling the function and showing the time to compute a specific metric and show that it is really fast. and add an additional section in the documentation to refer to this particular example if one is interested to know the details.

**Comment by auguste-probabl** [comment]
My misunderstanding here is about what is in `_cache`, so if we don't want to talk about `_cache` in the docs then this issue is kind of blocked.
